### PR TITLE
GIT-3005:Merge request merged -> transition an issue script was imple…

### PR DESCRIPTION
**Pull request by adokaeva**
Script was implemented in suggestion that JIRA Issue Key is present in merge request title.
First issue's key from merge/pull request title will be taken as issue for transition.
Issue's status will be changed on behalf of issue's assignee.

Please, take into account that Fix version retains empty.

### CODE REVIEW

File: scripts/on-merge-req-updated.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
+//# sourceURL=on-merge-req-created.js
+'use strict';
+//  Description
+//
+// This script work as a handler that checks if an issue is in in progress status.
```
Comment (dmalyshkin-issart) [inline scripts/on-merge-req-updated.js]: I've slightly re-phrased the description:

//  This script does an automatic issue transition to "Done" status if
// the issue is still in "In progress" status when a merge/pull request has been updated for the issue.
// It works on the assumption that an issue key is present in the title of the merge/pull request.
// First issue's key from the merge/pull request title will be taken as an issue for the transition.
// Issue's status will be changed on behalf of the issue's assignee.
// Please take into account that 'Fix version' field remains empty.

File: scripts/on-merge-req-updated.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+// https://github.com/BigBrassBand/jira-git-workflow-hooks
+//
+//  Applying
+// Set name of in progress status in constant IN_PROGRESS. By default "In Progress" is used.
+// Set name of needed transition in constant DONE. By default "Resolve Issue" is used.
+//
+//  Classes passed as parameters by default
+//
+//  A JSON object which contains the merge request data.
+// pullreqJson : org.json.JSONObject
```
Comment (dmalyshkin-issart) [inline scripts/on-merge-req-updated.js:22]: You have missed 'oldPullreqJson' parameter. It is not defined for 'on-merge-req-created.js', but present for 'on-merge-req-updated.js'.

File: scripts/on-merge-req-updated.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+//  Contains the next fields:
+//    "requestId"    -- ID of this merge request.
+//    "requestState" -- Text representation of the current request state.
```
Comment (dmalyshkin-issart) [inline scripts/on-merge-req-updated.js:25]: Does it make sense to check this 'requestState' field? I.e. to move the found issue to 'Done' status only when pull/merge request is resolved.
Current implementation makes transition on *ANY* pull/merge request update. For example, on request title change.

File: scripts/on-merge-req-updated.js (hunk 1 above)
Comment (adokaeva) [reply scripts/on-merge-req-updated.js]: done

File: scripts/on-merge-req-updated.js (hunk 2 above)
Comment (adokaeva) [reply scripts/on-merge-req-updated.js:22]: corrected

File: scripts/on-merge-req-updated.js (hunk 3 above)
Comment (adokaeva) [reply scripts/on-merge-req-updated.js:25]: Yes, It looks like you are right.
I have added this check.

**Pull request merged**